### PR TITLE
Implement ViewHolder for all adapter classes and integrate Butterknife where applicable

### LINKED ISSUE #85: Integrate butterknife everywhere
There are still a lot of pieces of code where butterknife is not integrated.
`FinalCityInfo`
`TweetsAdapter`
`PlacesOnMap`
`FunfactFragment`
`CityAdapter`
`Checklist`
`ChecklistFragment`
`ShareContact`
`MyTripFriendnameAdapter`
`MyTripInfoImagesAdapter`
`BusAdapter`
`MyTripInfo`
`Hotels`
`ShoppingAdapter`
`TrainAdapter`
`MapRealTimeActivity`
`TripImage`
`MyTripsAdapter`
Use butterknife in above files.

### PR DESCRIPTION
**Pull request by amy6**
# Description

* Implement ViewHolder for all adapter classes for improving the performance of Listviews by reducing the number of calls to `findViewById` method
* Integrate Butterknife to facilitate view bindings without explicitly calling `findViewById` methods

Fixes #85 and #95 

## Type of change
Just put an x in the [] which are valid.
- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

# How Has This Been Tested?
Tested on physical device (OP3 API 26)
Please describe the tests that you ran to verify your changes.
- [ ] `./gradlew assembleDebug assembleRelease`
- [ ] `./gradlew checkstyle`

# Checklist:
- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [x] My changes generate no new warnings

## Additional Notes to the reviewer:
* Commits have been made for group of classes together to be able to track the changes easily.
Kindly review the changes to **CityAdapter** class - have defined two `ViewHolder` classes to take care of a child view being inflated inside Adapter's `getView` method.

* Also I noticed that the Login UI has been modified to below:
travelmate-login
Checked the previous commits and I believe that this change was introduced in commit a0125feeabf00669b066bfc7459d9aea515b3895. Please take a look.

### CODE REVIEW

File: .idea/Travel-Mate.iml [hunk 1]
```diff
@@ -0,0 +1,9 @@
+<?xml version="1.0" encoding="UTF-8"?>
```
Comment (Swati4star) [inline .idea/Travel-Mate.iml]: remove this & next 3 files from commit

File: Android/app/src/main/java/io/github/project_travel_mate/travel/mytrips/MyTripsAdapter.java [hunk 2]
```diff
@@ -21,33 +21,41 @@
...
-    public View getView(final int position, View view2, @NonNull ViewGroup parent) {
-        LayoutInflater mInflater = (LayoutInflater) mContext.getSystemService(Activity.LAYOUT_INFLATER_SERVICE);
-        View view = Objects.requireNonNull(mInflater).inflate(R.layout.trip_listitem, parent, false);
-        ImageView city = view.findViewById(R.id.profile_image);
-        TextView cityname = view.findViewById(R.id.tv);
-        TextView date = view.findViewById(R.id.date);
+    public View getView(final int position, View convertView, @NonNull ViewGroup parent) {
+        ViewHolder holder;
+        View view = convertView;
+        if(view == null) {
```
Comment (Swati4star) [inline Android/app/src/main/java/io/github/project_travel_mate/travel/mytrips/MyTripsAdapter.java]: Add space after `if` to fix the checkstyle error

**Comment by Swati4star** [comment]
@amy6 The changes look good now. Just squash the commits into one now.

**Comment by amy6** [comment]
Finally, done with it @Swati4star. :sweat_smile: Squashed all of them as a single commit. Checkstyle error has been taken care of.

**Comment by Swati4star** [comment]
Merged in d99e0bb2b4a3c82c7bf611283b4f44fe899c4231

Thanks @amy6 👍

**Pull request closed without merging**